### ISSUE TITLE
Conversion from 'AssemblyScript' to 'Value' for source type 'Array<ethereum.Tuple>' is not supported - graph-cli 0.56.0

### ISSUE THREAD

**Issue by 0xMikado** [graphprotocol/graph-tooling#1429]
### Which packages are impacted by your issue?

@graphprotocol/graph-cli

### Describe the issue

Hello, it appears graph codegen still fails for an event/function with a Tuple[] structure. 

It has been discussed a lot of times in different issues, with apparently a fix in some of the graph cli versions. You can see previous discussions on this here: 
- [issue#342](https://github.com/graphprotocol/graph-tooling/issues/342) 
- [issue#253](https://github.com/graphprotocol/graph-tooling/issues/253)  
- [pull#456](https://github.com/graphprotocol/graph-tooling/pull/465)

However, I still face this issue on the latest version (0.56.0).

### Reproduction

no

### Steps to Reproduce the Bug or Issue

Run graph add 0x0259119359Bf053ebF42C9807752de6bbb4925f3
You should get the following error: 
Error: Conversion from 'AssemblyScript' to 'Value' for source type 'Array<ethereum.Tuple>' is not supported

### Expected behavior

To have graph add run properly.

### Platform

- OS: [macOS]
- NodeJS: [18.12.0]
- @graphprotocol/graph-cli: [0.56.0]
- @graphprotocol/graph-ts: [0.31.0],

**Comment by dimitrovmaksim** [comment]
I believe this happens when `init` or `add` try go generate an example entity in the schema based on an event that has a struct/list of structs as a parameter.

**Comment by jhubbardsf** [comment]
Yeah, I'm seeing this right now with a struct being a param in an event. DiamondCut_diamondCutStruct trying to be assigned to Bytes.

**Comment by azf20** [comment]
thanks @0xMikado and @dimitrovmaksim - do you think this is a regression or was this case never fixed? @saihaj do you have an idea where this might be tripping up?

**Comment by dimitrovmaksim** [comment]
I think this was never fixed. And i'm not sure it could be fixed for all contracts. 

If the type of the event param is `ethereum.Tuple` / `ethereum.Tuple[]`, it probably means it's a struct/array of structs. In newer ABIs  (can't remember the solidity version where it was introduced), event inputs have a property `internalType` that will hold the name of the struct. One way to fix this is to check if the input is actually a struct, then fetch the struct definition and it's properties from the abi and construct an Entity based on that struct. But this property `internalType` is not present in older abis

**Comment by azf20** [comment]
ok - in that case seems like a partial fix is the best possible option, and at least could provide better error messages in that case
